llama_stack/apis/common/type_system.py

```diff
@@ -46,6 +46,21 @@ class CustomType(BaseModel):
     validator_class: str
 
 
+class ChatCompletionInputType(BaseModel):
```

Comment:
Why create these new types?

Reply:
@raghotham because they are sub-types of the ChatCompletionRequest? not all fields are expected to be in the input row. Some fields like model and sampling_params are part of the "EvalCandidate". Does that make sense?

Reply:
The client will need to specify the dataset schema: 

Option 1: 
```
dataset_schema = {
   # validate against messages=List[Message]
   # equivalent to {"type": "custom", "validator_class": "ChatCompletionInput"}
   "chat_completion_input": CustomType(validator_class="ChatCompletionInput")
}
```

Option 2: 
```
dataset_schema = {
   # validate against messages=List[Message]
   "chat_completion_input": ChatCompletionInputType(), # {"type": "chat_completion_input"}
}
```

These new types are added given we are going with Option 2 from offline discussion w/ @ashwinb